Makefile

```diff
@@ -37,13 +37,21 @@ test-unit:
 # Run the code generation tasks.
 # Example:
 #    make update
-update: install-go-bindata
+update: install-go-bindata install-go-deps
 	hack/update-codegen.sh
 	hack/update-generated-bindata.sh
 
 install-go-bindata:
 	hack/install-go-bindata.sh
 
+install-go-deps:
+	go mod tidy
+	rm -rf ./vendor
```

Comment:
as discussed in chat, we shouldn't be doing this but all this should be idempotent (like `dep ensure` today)

Reply:
https://github.com/openshift/cluster-monitoring-operator/blob/master/Makefile#L76

Reply:
I'm also getting this diff when just running `go mod tidy` or `vendor`

Reply:
if it is indeed idempotent, this should be safe to do though, right?
Remove anyway?

Reply:
which diff are we talking about?
Should this just be:
```
go mod tidy
go mod vendor
go mod verify
```

Reply:
> if it is indeed idempotent, this should be safe to do though, right?
> Remove anyway?

it will re-download everything if you remove - if you don't it'll probably just do an hash check and prevents downloading if they match

> which diff are we talking about?

when I run `go mod tidy` or `go mod verify` (sorry haven't pasted the diff lmao), I get this from this PR in my local development workspace:

```diff
diff --git a/go.mod b/go.mod
index 2205020f..7f2bf48f 100644
--- a/go.mod
+++ b/go.mod
@@ -13,7 +13,6 @@ require (
        github.com/Microsoft/hcsshim v0.8.6 // indirect
        github.com/ajeddeloh/go-json v0.0.0-20170920214419-6a2fe990e083 // indirect
        github.com/ajeddeloh/yaml v0.0.0-20170912190910-6b94386aeefd // indirect
-       github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf // indirect
        github.com/apparentlymart/go-cidr v1.0.0
        github.com/ashcrow/osrelease v0.0.0-20180626175927-9b292693c55c
        github.com/containerd/cgroups v0.0.0-20190710130057-d596c78861b1 // indirect
@@ -49,7 +48,6 @@ require (
        github.com/joho/godotenv v1.3.0
        github.com/konsorten/go-windows-terminal-sequences v1.0.2 // indirect
        github.com/kubernetes-sigs/cri-o v1.12.10
-       github.com/modern-go/reflect2 v1.0.1 // indirect
        github.com/mtrmac/gpgme v0.0.0-20170102180018-b2432428689c // indirect
        github.com/opencontainers/go-digest v1.0.0-rc1
        github.com/opencontainers/image-spec v1.0.1 // indirect
```

Reply:
talking about idempotency, it doesnt seem to do that on my system :smile: